replication project.Rmd

```diff
@@ -0,0 +1,259 @@
+estmean <- model$coef$fixed # Estimated means for each variable 
+estvar <- vcov(model) # Estimated covariances
+se <- rep(NA, length(variable)) # Creating empty vector for predicted s.e. values
+# Creating a loop using the deltamethod function which approximates standard errors of transformation of parameters based on the given estimates of the mean and covariance (~ bootstrapping).
+for (i in 1:length(variable)) {
+  j <- variable[i]
+  se[i] <- deltamethod (~ (x17) + (x36) * j, estmean, estvar)
+}
+
+# Create confidence intervals and combine all values in matrix.
+upper <- slopes + 1.96 * se
+lower <- slopes - 1.96 * se
+variable.data <- cbind(variable, slopes, upper, lower)
+
+# Creating the first graph with ggplot
+
+variable.data <- data.frame(variable.data) # Turn matrix into data frame so that ggplot can read it.
+xtitle <- "Asylumseekers"
+ytitle <- "Effect: Unemployment"
+main <- "Figure 1: The Conditional Effects of Unemployment and Immigration"
+graph_1 <- ggplot(variable.data, aes(x = variable, y = slopes)) + 
+  geom_line(aes(y = slopes), color = "blue") + 
+  geom_ribbon(aes(ymin = lower, ymax = upper), alpha = 0.2) + 
+  geom_hline(yintercept = 0, size = 0.4, color = "black") +
+  labs(x = xtitle, y = ytitle, title = main) +
+  coord_fixed(ratio = 8)
+
+#Figure 1, graph 2
+
+unemployment <- seq(from = min(data$z_unemp), to = max(data$z_unemp), by = .01) # vector with possible values for unemployment
+variable <- unemployment # standardized label
+slopes <- model$coef$fixed["z_asylumseeker"] + model$coef$fixed["z_asylumseeker:z_unemp"] * variable # Estimated slopes based on estimated coefficients of asylumseekers (16) and unemployment x immigration (36)
+estmean <- model$coef$fixed # Estimated means for variables 
+estvar <- vcov(model) # Estimated covariances
+se <- rep(NA, length(variable)) # Creating empty vector for predicted s.e. values
+# Creating a loop with deltamethod to calculate estimated standard errors
+for (i in 1:length(variable)) {
+  j <- variable[i]
+  se[i] <- deltamethod (~ (x16) + (x36) * j, estmean, estvar)
+}
+# Create confidence intervals and combine all values in matrix.
+upper <- slopes + 1.96 * se
+lower <- slopes - 1.96 * se
+variable.data <- cbind(variable, slopes, upper, lower)
+
+# Creating the second graph with ggplot:
+
+variable.data <- data.frame(variable.data) # Turn matrix into data frame so that ggplot can read it.
+xtitle <- "Unemployment"
+ytitle <- "Effect: Asylum seekers"
+main <- "The Conditional Effects of Unemployment and Immigration"
+graph_2 <- ggplot(variable.data, aes(x = variable, y = slopes)) + 
+  geom_line(aes(y = slopes), color = "blue") + 
+  geom_ribbon(aes(ymin = lower, ymax = upper), alpha = 0.2) + 
+  geom_hline(yintercept = 0, size = 0.4, color = "black") +
+  labs(x = xtitle, y = ytitle) +
+  coord_fixed(ratio = 12)
+
+# Figure 1 with ggplot
+
+grid.arrange(graph_1, graph_2, ncol = 1)
+
+# Plot separately - better scaled
+graph_1
+graph_2
+
```

Comment:
the code to create these graphs is very similar.  If you want you could turn it into a function.  Then you could create as many plots as you want like this.